src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,851 @@
+            If ``pd.DataFrame`` is passed, geometry column must
+            have the name ``geometry``. Passed to
+            ``aat.compute_raster_stats()``.
+        feature_col : str
+            Column in ``gdf`` to use as row/feature identifier.
+            and dates. Passed to ``aat.compute_raster_stats()``.
+            The string is also used as a suffix to the
+            processed file path for unique identication of
+            analyses done on different files and columns.
+        clobber : bool, default = False
+            If True, overwrites existing processed dates. If
+            the new file matches the old file, dates will be
+            reprocessed and appended to the data frame. If
+            files do not match, the old file will be replaced.
+            If False, stats are only calculated for year-dekads
+            that have not already been calculated within the
+            file. However, if False and files do not match,
+            value error will be raised.
+        **kwargs
+            Additional keyword arguments passed to
+            ``aat.computer_raster_stats()``.
+
+        Returns
+        -------
+        Path
+            The processed path
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, \
+        ...  CodAB, UsgsNdviSmoothed
+        >>>
+        >>> # Retrieve admin 2 boundaries for Burkina Faso
+        >>> country_config = create_country_config(iso3="bfa")
+        >>> codab = CodAB(country_config=country_config)
+        >>> bfa_admin2 = codab.load(admin_level=2)
+        >>> bfa_admin1 = codab.load(admin_level=1)
+        >>>
+        >>> # setup NDVI
+        >>> bfa_ndvi = UsgsNdviSmoothed(
+        ...     country_config=country_config,
+        ...     start_date=[2020, 1],
+        ...     end_date=[2020, 3]
+        ... )
+        >>> bfa_ndvi.download()
+        >>> bfa_ndvi.process(
+        ...     gdf=bfa_admin2,
+        ...     feature_col="ADM2_FR"
+        ... )
+        >>>
+        >>> # process for admin1
+        >>> bfa_ndvi.process(
+        ...     gdf=bfa_admin1,
+        ...     feature_col="ADM1_FR"
+        ... )
+        """
+        # get statistics for processing
+        stats_list = kwargs.pop(
+            "stats_list", ["mean", "std", "min", "max", "sum", "count"]
+        )
+        percentile_list = kwargs.pop("percentile_list", None)
+
+        if not stats_list and not percentile_list:
+            raise ValueError(
+                "At least one of `stats_list` or "
+                "`percentile_list` must not be `None` "
+                "when passed to `process()`."
+            )
+
+        # get dates for processing
+        all_dates_to_process = _expand_dekads(
+            y1=self._start_year,
+            d1=self._start_dekad,
+            y2=self._end_year,
+            d2=self._end_dekad,
+        )
+
+        if stats_list is not None:
```

Comment:
I think it would be simple to create a combined stats / percentile list here:
`full_stats_list = stats_list + [f"{p}quant" for p in percentile_list]`
Then you can simply loop through this, and pass it to a single parameter called `stat` in `_process`

Reply:
Then I guess in `_process()` we reprocess the stat passed in to see if it's a percentile or a stat, so which way to pass to the `compute_raster_stats()`. Maybe this is a longer term issue that we sort out by changing the parameters passed to `compute_raster_stats()`? As it stands, you have to separately pass the stats or percentile.

Reply:
Ah I'm so sorry, my bad again, I thought that `_process` didn't differentiate between a normal stat and percentile after appending `quant`, but indeed it's important for passing to `compute_raster_stats`. 

A cleaner way to do this might be to just have an arg for a single stat (as a string), and another arg for a boolean that specifies `is_percentile`. But it's also fine the way it is.

Reply:
Love the idea, have changed!